lib/generators/json_serializable_generator.dart

```diff
@@ -192,6 +216,24 @@ bool _hasFromJsonCtor(DartType type) {
   return false;
 }
 
+DartType _listTypeWithFromJsonCtor(DartType type) {
+  if (type is! InterfaceType) return null;
+  if (!_isDartList(type)) return null;
+
+  var listType = (type as InterfaceType).typeArguments.first;
```

Comment:
added a test for a List<dynamic> which it just leaves alone like all lists were left before this change